engine/src/main/resources/application.conf

```diff
@@ -72,85 +65,121 @@ call-caching {
-  // This allows cromwell to localize data that wouldn't be accessible using whatever "cromwellAuthenticationScheme" has been defined.
-  // The currently only available scheme is refresh, which requires a refresh token to be passed upon every workflow submission (see "workflow options" in cromwell documentation).
-  // In that scheme, a client_id and client_secrets must be provided. They must be the ones that have been used to generate the above-mentioned refresh token.
-  // userAuthenticationScheme = "refresh"
-  // refreshTokenAuth = {
-  //   client_id = ""
-  //   client_secret = ""
-  // }
+backend {
+  default = "Local"
+  providers = [
+    {
+      name = "Local"
+      class = "cromwell.engine.backend.local.LocalBackend"
+      config {
+        // Root directory where Cromwell writes job results.  This directory must be
+        // visible and writeable by the Cromwell process as well as the jobs that Cromwell
+        // launches.
+        root: "cromwell-executions"
+
+        // Cromwell makes a link to your input files within <root>/<workflow UUID>/workflow-inputs
+        // The following are strategies used to make those links.  They are ordered.  If one fails
+        // The next one is tried:
+        //
+        // hard-link: attempt to create a hard-link to the file
+        // copy: copy the file
+        // soft-link: create a symbolic link to the file
+        //
+        // NOTE: soft-link will be skipped for Docker jobs
+        localization: [
+          "hard-link", "soft-link", "copy"
+        ]
+      }
+    },
+    {
+      name = "SGE"
+      class = "cromwell.engine.backend.sge.SgeBackend"
+      config {
+        // Root directory where Cromwell writes job results.  This directory must be
+        // visible and writeable by the Cromwell process as well as the jobs that Cromwell
+        // launches.
+        root: "cromwell-executions"
+
+        // Cromwell makes a link to your input files within <root>/<workflow UUID>/workflow-inputs
+        // The following are strategies used to make those links.  They are ordered.  If one fails
+        // The next one is tried:
+        //
+        // hard-link: attempt to create a hard-link to the file
+        // copy: copy the file
+        // soft-link: create a symbolic link to the file
+        //
+        // NOTE: soft-link will be skipped for Docker jobs
+        localization: [
+          "hard-link", "soft-link", "copy"
+        ]
+      }
+    },
+    {
+      name = "JES"
+      class = "cromwell.engine.backend.jes.JesBackend"
+      config {
+        // Google project
+        project = "my-cromwell-workflows"
+
+        // Base bucket for workflow executions
+        root = "gs://my-cromwell-workflows-bucket"
+
+        // Endpoint for APIs, no reason to change this unless directed by Google.
+        endpointUrl = "https://genomics.googleapis.com/"
```

Comment:
This is specific of JES backend implementation and there is no need to generalized. I would recommend to put this in other section specific for JES backend implementation.